### PR TITLE
[nrf fromtree] Python tool version change: Breathe & Spinx

### PR DESCRIPTION
**Pull request by thst-nordic**
https://github.com/zephyrproject-rtos/zephyr/pull/48947
    Update to Sphinx 5.x, also update breathe since version 4.34 is the
    first to officially support Sphinx 5.x.
    
    Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>
    Signed-off-by: Thomas Stilwell <Thomas.Stilwell@nordicsemi.no>

### CODE REVIEW

**Review by rlubos — CHANGES_REQUESTED** [review]
The sauce tag for the no up commit should be `[nrf noup]`. Also please fix the commit message format to pass the compliance.

**Comment by rlubos** [comment]
And we need a manifest update PR

**Comment by thst-nordic** [comment]
i have the sauce typo fix ready to push, but waiting for the doc build action to see if any more issues.

**Comment by thst-nordic** [comment]
waiting for +1 from @gmarull

**Review by rlubos — APPROVED** [review]

File: .github/workflows/doc-build.yml [hunk 1]
```diff
@@ -89,7 +89,7 @@ jobs:
           DOC_TARGET="html"
         fi
 
-        DOC_TAG=${DOC_TAG} SPHINXOPTS="-q -W -t publish" make -C doc ${DOC_TARGET}
```
Comment (gmarull) [inline .github/workflows/doc-build.yml:92]: We have no warnings upstream, so why is this required?

**Review by gmarull — CHANGES_REQUESTED** [review]

**Review by tejlmand — APPROVED** [review]
lgtm

File: .github/workflows/doc-build.yml (hunk 1 above)
Comment (thst-nordic) [reply .github/workflows/doc-build.yml:92]: you can see the discussion about it here: NCSDK-16980

Comment (thst-nordic) [reply .github/workflows/doc-build.yml]: done
